src/HotChocolate/Core/test/Types.Tests/Types/Relay/NodeTypeTests.cs

```diff
@@ -243,6 +243,43 @@ async Task Error() => await new ServiceCollection()
         error.Message.MatchSnapshot();
     }
 
+    [Fact]
+    public async Task Infer_Node_From_Query_As_Interface_From_Interface()
+    {
+        await new ServiceCollection()
+            .AddGraphQL()
+            .AddQueryType<Query10>()
+            .AddType<Query10.DrivingLicense>() // TODO: Should this be discovered?
```

Comment:
// TODO: Should this be discovered?

Reply:
I think since we will allow `abstract / interface` types to be used on node resolvers, we should automatically discover and register all the derived types on the schema. What do you think?